src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/Extensions/ContextQuery/SyntaxTreeExtensions.vb

```diff
@@ -354,6 +354,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions.ContextQuery
                 Return False
             End If
 
+            Dim directive As DirectiveTriviaSyntax = Nothing
+            If IsInPreprocessorDirectiveContext(syntaxTree, position, cancellationToken, directive) AndAlso
+               directive.IsKind(SyntaxKind.IfDirectiveTrivia, SyntaxKind.ElseIfDirectiveTrivia) Then
+                Return False
+            End If
```

Comment:
this matches our C# logic. we don't consider a pp-directive to be a general expression context.